provider/openrouter/openrouter.go

```diff
@@ -82,6 +108,50 @@ func (p *Provider) Models() (models []model.Model, err error) {
 	return models, nil
 }
 
+// providerModels returns the provider's list of models given the URL to fetch the models.
+func providerModels(url string) (models ModelsList, err error) {
+	request, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return ModelsList{}, pkgerrors.WithStack(err)
+	}
+	request.Header.Set("Accept", "application/json")
+
+	client := &http.Client{}
+	if err := retry.Do( // Query available models with a retry logic cause "openrouter.ai" has failed us in the past.
+		func() error {
+			response, err := client.Do(request)
+			if err != nil {
+				return pkgerrors.WithStack(err)
+			}
+			defer response.Body.Close()
+
+			responseBody, err := io.ReadAll(response.Body)
+			if err != nil {
+				return pkgerrors.WithStack(err)
+			}
+
+			if response.StatusCode != http.StatusOK {
+				return pkgerrors.Errorf("received status code %d when querying provider models", response.StatusCode)
+			}
```

Comment:
This is still verifying the response, so please move it up to where the response is handled.

Reply:
Below the `defer`.